### ISSUE TITLE
[Request] Source migration

### ISSUE THREAD

**Issue by Lustsifer** [LagradOst/QuickNovel#31]
A features to quickly migrate a novel from source A to source B. 

![Screenshot_20210926-144844.png](https://user-images.githubusercontent.com/84452966/134798838-b3159785-2407-4418-8300-572b9f867585.png)
![Screenshot_20210926-144908.png](https://user-images.githubusercontent.com/84452966/134798840-770ace0b-cdb9-4c11-a24c-16f46eb0225e.png)
![Screenshot_20210926-145015.png](https://user-images.githubusercontent.com/84452966/134798841-7427e910-a073-4f88-866b-083413d58356.png)

**Comment by LagradOst** [comment]
Why?

**Comment by CloudS33ker** [comment]
It's an excellent idea for me. Some novels aren't religiously updated on some websites. One website might have 100 chapters of a novel while the other would have 300 plus and regularly updating. 

Also, sometimes a website is unavailable due to various reasons, it's always better to have that backup option.

I have no idea how it'll be implemented though. There are discrepancies in the chapter listings among these websites and chapter number 26 might be referred to as chapter number 28 in another website.

**Comment by LagradOst** [comment]
@CloudS33ker yes, this is the main problem, as all sites use a slightly different name for chapters and some chapters are missing causing missing chapters or jump (also the download location is fucked if I link providers this way)

TLDR: hard to program with the current system and hard to make a working solution.

My recommendation is to redownload it from another provider and that is what I have done for some stuff. Not the best, but it works when you find a reliable provider.

**Comment by CloudS33ker** [comment]
Can listings be merged though? 
Like if I open the info page of one novel and it shows 3 different tabs (or drop down list options) for 3 different sources or something. I've seen it happen for one of the manga reader apps long ago. The actual chapter list might be different for each website but they'll all show up linked to the same info page.

I know even this will run into problems when websites don't label the novels properly but it should work for most.

**Comment by LagradOst** [comment]
> 
> 
> Can listings be merged though? Like if I open the info page of one novel and it shows 3 different tabs (or drop down list options) for 3 different sources or something. I've seen it happen for one of the manga reader apps long ago. The actual chapter list might be different for each website but they'll all show up linked to the same info page.
> 
> I know even this will run into problems when websites don't label the novels properly but it should work for most.

So add a button to show the search results on all providers given the title of the current novel?

**Comment by Lustsifer** [comment]
> So add a button to show the search results on all providers given the title of the current novel?

How about a feature to manually merge them into one? I remember it called that on Tachiyomi. You can merge novel A from source 1,2,3,etc. into one.

**Comment by CloudS33ker** [comment]
Both could work actually. Both have their distinct advantages too. 

Final decision rests with you though. Whichever is more feasible to implement.
